js/src/components/enhanced-conversion-tracking-settings/cta.test.js

```diff
@@ -0,0 +1,221 @@
+} ) );
+
+jest.mock( '.~/hooks/useAcceptedCustomerDataTerms', () => ( {
+	__esModule: true,
+	default: jest.fn().mockName( 'useAcceptedCustomerDataTerms' ),
+} ) );
+
+jest.mock( '.~/hooks/useAllowEnhancedConversions', () => ( {
+	__esModule: true,
+	default: jest.fn().mockName( 'useAllowEnhancedConversions' ),
+} ) );
+
+jest.mock( '.~/hooks/useAutoCheckEnhancedConversionTOS', () => ( {
+	__esModule: true,
+	default: jest
+		.fn()
+		.mockName( 'useAutoCheckEnhancedConversionTOS' )
+		.mockImplementation( () => {
+			return {
+				startEnhancedConversionTOSPolling: jest.fn(),
+				stopEnhancedConversionTOSPolling: jest.fn(),
+			};
+		} ),
+} ) );
+
+jest.mock( '.~/data/actions', () => ( {
+	...jest.requireActual( '.~/data/actions' ),
+	updateEnhancedAdsConversionStatus: jest
+		.fn()
+		.mockName( 'updateEnhancedAdsConversionStatus' )
+		.mockImplementation( () => {
+			return { type: 'test', response: 'enabled' };
+		} ),
+} ) );
+
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+
+/**
+ * Internal dependencies
+ */
+import useAcceptedCustomerDataTerms from '.~/hooks/useAcceptedCustomerDataTerms';
+import useAllowEnhancedConversions from '.~/hooks/useAllowEnhancedConversions';
+import { ENHANCED_ADS_CONVERSION_STATUS } from '.~/constants';
+import CTA from './cta';
+
+describe( 'Enhanced Conversion CTA', () => {
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	test( 'Prompt the user to sign the TOS', () => {
+		useAcceptedCustomerDataTerms.mockReturnValue( {
+			acceptedCustomerDataTerms: false,
+			hasFinishedResolution: true,
+		} );
+
+		useAllowEnhancedConversions.mockReturnValue( {
+			allowEnhancedConversions: null,
+		} );
+
+		render( <CTA /> );
+		expect(
+			screen.getByText( 'Accept Terms & Conditions' )
+		).toBeInTheDocument();
+	} );
+
+	test( 'Prompt the user to enable enhanced conversion tracking if the TOS has been accepted', () => {
+		useAcceptedCustomerDataTerms.mockReturnValue( {
+			acceptedCustomerDataTerms: true,
+			hasFinishedResolution: true,
+		} );
+
+		useAllowEnhancedConversions.mockReturnValue( {
+			allowEnhancedConversions: ENHANCED_ADS_CONVERSION_STATUS.DISABLED,
+		} );
+
+		render( <CTA enableLabel="Confirm" /> );
+		expect( screen.getByText( 'Confirm' ) ).toBeInTheDocument();
+	} );
+
+	test( 'Prompt the user to disable enhanced conversion tracking if enabled', () => {
+		useAcceptedCustomerDataTerms.mockReturnValue( {
+			acceptedCustomerDataTerms: true,
+			hasFinishedResolution: true,
+		} );
+
+		useAllowEnhancedConversions.mockReturnValue( {
+			allowEnhancedConversions: ENHANCED_ADS_CONVERSION_STATUS.ENABLED,
+		} );
+
+		render( <CTA disableLabel="Disable tracking" /> );
+		expect( screen.getByText( 'Disable tracking' ) ).toBeInTheDocument();
+	} );
+
+	test( 'Click on accept TOS button callback', () => {
+		window.open = jest.fn();
```

Comment:
Consider `window` is a global object across jest environment, it could be mocked by

```js
const spy = jest.spyOn( global.window, 'open' ).mockImplementation();

expect( spy ).toHaveBeenCalledTimes( 1 );

// Restore it in the end of this test case.
spy.mockRestore();
```